repo/fsrepo/migrations/ipfsdir.go

```diff
@@ -0,0 +1,129 @@
+package migrations
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"strconv"
+	"strings"
+
+	"github.com/mitchellh/go-homedir"
+)
+
+const (
+	envIpfsPath = "IPFS_PATH"
+	defIpfsDir  = ".ipfs"
+	versionFile = "version"
+
+	// Local IPFS API
+	apiFile = "api"
+)
+
+func init() {
+	homedir.DisableCache = true
+}
+
+// ApiEndpoint reads the api file from the local ipfs install directory and
+// returns the address:port read from the file.  If the ipfs directory is not
+// specified then the default location is used.
+func ApiEndpoint(ipfsDir string) (string, error) {
+	ipfsDir, err := CheckIpfsDir(ipfsDir)
+	if err != nil {
+		return "", err
+	}
+	apiPath := path.Join(ipfsDir, apiFile)
+
+	apiData, err := ioutil.ReadFile(apiPath)
```

Comment:
@Stebalien, this seems fine to me. Is it reasonable to ignore the reasoning behind the limiting here? https://github.com/ipfs/go-ipfs/blob/3393b4a39efae379daed421ec1754499affe849f/repo/fsrepo/fsrepo.go#L338